pkg/controller/plan/validation.go

```diff
@@ -114,6 +114,15 @@ func (r *Reconciler) validate(plan *api.Plan) error {
 	plan.Referenced.Provider.Source = pv.Referenced.Source
 	plan.Referenced.Provider.Destination = pv.Referenced.Destination
 
+	if plan.Referenced.Secret == nil &&
```

Comment:
wouldn't plan.Referenced.Secret always be nil when we get here?
additionally, if we don't use the secret in the validations (we shouldn't - the provider's state should cover the connectivity with the credentials within the secret) then why not placing it in the execute function? we'd save a call in case some validation fails..

Reply:
what execute function? we need this for the validation

Reply:
ah for the validation of the openshift cluster?

Reply:
ok, we still trigger a possibly redundant call here but at least we fixed a possible sig fault when the source provider is not set here - that's good enough